pkg/scheduler/plugins/loadaware/load_aware.go

```diff
@@ -188,64 +237,62 @@ func (p *Plugin) Score(ctx context.Context, state *framework.CycleState, pod *co
 		return 0, nil
 	}
 
+	prodPod := extension.GetPriorityClass(pod) == extension.PriorityProd
+	podMetrics := buildPodMetricMap(p.podLister, nodeMetric, prodPod)
+
 	estimatedUsed := estimatedPodUsed(pod, p.args.ResourceWeights, p.args.EstimatedScalingFactors)
-	estimatedAssignedPodUsage := p.estimatedAssignedPodUsage(nodeName, nodeMetric)
-	for resourceName, value := range estimatedAssignedPodUsage {
+	estimatedAssignedPodUsed, estimatedPods := p.estimatedAssignedPodUsed(nodeName, nodeMetric, podMetrics, prodPod)
+	for resourceName, value := range estimatedAssignedPodUsed {
 		estimatedUsed[resourceName] += value
 	}
-
-	allocatable := make(map[corev1.ResourceName]int64)
-	for resourceName := range p.args.ResourceWeights {
-		quantity := node.Status.Allocatable[resourceName]
-		if resourceName == corev1.ResourceCPU {
-			allocatable[resourceName] = quantity.MilliValue()
-		} else {
-			allocatable[resourceName] = quantity.Value()
+	if prodPod {
+		for resourceName, quantity := range sumPodUsages(podMetrics, estimatedPods) {
+			estimatedUsed[resourceName] += getResourceValue(resourceName, quantity)
 		}
+	} else {
 		if nodeMetric.Status.NodeMetric != nil {
```

Comment:
some pod usage may be counted repeatedly.

Reply:
resolved.